packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -41,20 +43,19 @@ export default function (server: Server, ctx: AppContext) {
         .select('did')
         .where(userLookupCol, '=', actor)
         .limit(1)
-      const mutedDidsQb = db
-        .selectFrom('mute')
-        .select('did')
-        .where('mutedByDid', '=', requester)
 
       let feedItemsQb = feedService
         .selectFeedItemQb()
         .where('originatorDid', '=', actorDidQb)
-        .where((qb) => {
+        .where((qb) =>
           // Hide reposts of muted content
-          return qb
-            .where('type', '!=', 'repost')
-            .orWhere('post.creator', 'not in', mutedDidsQb)
-        })
+          qb
+            .where('type', '=', 'post')
+            .orWhereNotExists(
+              accountService.mutedQb(requester, [ref('post.creator')]),
+            ),
+        )
+        .whereNotExists(graphService.blockQb(requester, [ref('post.creator')]))
```

Comment:
Is the block clause needed here?  I wonder if we may be able to pass on it, since a block between the two users is checked proactively before fetching the author feed.  Or maybe we only need to apply it to reposts, similar to mutes.

Reply:
this is actually for the case of a repost of a blocked user in another user's feed